PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = 'account.bank.statement.line'
-    _inherit = 'hr.employee'
-    _inherit = 'pos.config'
-    _inherit = "pos.order"
-    _inherit = "pos.payment"
-    _inherit = 'pos.session'
-    _inherit = 'product.product'
-    _inherit = 'res.config.settings'

Total Changes: 25

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/pos_hr/models/account_bank_statement.py
+++ b/addons/pos_hr/models/account_bank_statement.py
@@ -3,6 +3,6 @@ from odoo import fields, models
 
 
 class AccountBankStatementLine(models.Model):
-    _inherit = 'account.bank.statement.line'
+    _inherit = ['account.bank.statement.line']
 
     employee_id = fields.Many2one('hr.employee', string="Employee", help="The employee who made the cash move.")

--- a/addons/pos_hr/models/hr_employee.py
+++ b/addons/pos_hr/models/hr_employee.py
@@ -7,8 +7,9 @@ from odoo import api, models, _
 from odoo.exceptions import UserError
 from odoo.tools import format_list
 
+
 class HrEmployee(models.Model):
-    _inherit = 'hr.employee'
+    _inherit = ['hr.employee']
 
     @api.model
     def _load_pos_data_domain(self, data):

--- a/addons/pos_hr/models/multi_employee_sales_report.py
+++ b/addons/pos_hr/models/multi_employee_sales_report.py
@@ -2,8 +2,7 @@
 from odoo import api, models
 
 
-class MultiEmployeeSalesReport(models.AbstractModel):
-    _name = 'report.pos_hr.multi_employee_sales_report'
+class ReportPos_HrMulti_Employee_Sales_Report(models.AbstractModel):
     _description = 'A collection of single session reports. One for each employee'
 
     @api.model

--- a/addons/pos_hr/models/pos_config.py
+++ b/addons/pos_hr/models/pos_config.py
@@ -4,7 +4,7 @@ from odoo import models, fields, api
 
 
 class PosConfig(models.Model):
-    _inherit = 'pos.config'
+    _inherit = ['pos.config']
 
     basic_employee_ids = fields.Many2many(
         'hr.employee', 'pos_hr_basic_employee_hr_employee', string="Employees with basic access",

--- a/addons/pos_hr/models/pos_order.py
+++ b/addons/pos_hr/models/pos_order.py
@@ -4,7 +4,7 @@ from markupsafe import Markup
 
 
 class PosOrder(models.Model):
-    _inherit = "pos.order"
+    _inherit = ["pos.order"]
 
     employee_id = fields.Many2one('hr.employee', string="Cashier", help="The employee who uses the cash register.")
     cashier = fields.Char(string="Cashier name", compute="_compute_cashier", store=True)

--- a/addons/pos_hr/models/pos_payment.py
+++ b/addons/pos_hr/models/pos_payment.py
@@ -2,7 +2,7 @@ from odoo import models, fields, api
 
 
 class PosPayment(models.Model):
-    _inherit = "pos.payment"
+    _inherit = ["pos.payment"]
 
     employee_id = fields.Many2one('hr.employee', string='Cashier', related='pos_order_id.employee_id', store=True, index=True)
 

--- a/addons/pos_hr/models/pos_session.py
+++ b/addons/pos_hr/models/pos_session.py
@@ -5,7 +5,7 @@ from odoo.tools import plaintext2html
 
 
 class PosSession(models.Model):
-    _inherit = 'pos.session'
+    _inherit = ['pos.session']
     employee_id = fields.Many2one(
         "hr.employee",
         string="Cashier",

--- a/addons/pos_hr/models/product_product.py
+++ b/addons/pos_hr/models/product_product.py
@@ -2,7 +2,7 @@ from odoo import api, models
 
 
 class ProductProduct(models.Model):
-    _inherit = 'product.product'
+    _inherit = ['product.product']
 
     @api.model
     def _load_pos_data_fields(self, config_id):

--- a/addons/pos_hr/models/res_config_settings.py
+++ b/addons/pos_hr/models/res_config_settings.py
@@ -4,7 +4,7 @@ from odoo import fields, models, api
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     # pos.config fields
     pos_basic_employee_ids = fields.Many2many(related='pos_config_id.basic_employee_ids', readonly=False,

--- a/addons/pos_hr/models/single_employee_sales_report.py
+++ b/addons/pos_hr/models/single_employee_sales_report.py
@@ -3,9 +3,8 @@ from odoo import api, models
 from odoo.osv.expression import AND
 
 
-class SingleEmployeeSalesReport(models.AbstractModel):
-    _name = 'report.pos_hr.single_employee_sales_report'
-    _inherit = 'report.point_of_sale.report_saledetails'
+class ReportPos_HrSingle_Employee_Sales_Report(models.AbstractModel):
+    _inherit = ['report.point_of_sale.report_saledetails']
     _description = 'Session sales details for a single employee'
 
     def _get_domain(self, date_start=False, date_stop=False, config_ids=False, session_ids=False, employee_id=False):
